### PR TITLE
Navigation drawer

### PR DESCRIPTION
**Pull request by falseinput**

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -12,12 +12,12 @@ import RegisterPage from './src/javascripts/pages/register/register-page';
...
 import store from './src/javascripts/redux/store';
 import AddLocationPage from "./src/javascripts/pages/add-location/add-location-page";
-
+import SideMenu from './src/javascripts/pages/side-menu/side-menu';
 
 // To see all the requests in the chrome Dev tools in the network tab.
 XMLHttpRequest = GLOBAL.originalXMLHttpRequest ?
-    GLOBAL.originalXMLHttpRequest :
-    GLOBAL.XMLHttpRequest;
+  GLOBAL.originalXMLHttpRequest :
```
Comment (falseinput) [inline App.js]: Nice catch

File: src/javascripts/pages/side-menu/side-menu.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { NavigationActions } from 'react-navigation';
+import { AsyncStorage, ScrollView, Text, View, ToastAndroid } from 'react-native';
+import STRINGS from '../../values/strings';
+import styles from './styles';
+import userActions from '../../redux/actions/user';
+
```
Comment (falseinput) [inline src/javascripts/pages/side-menu/side-menu.js]: Two newlines between imports and content

File: src/javascripts/pages/side-menu/side-menu.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+class SideMenu extends React.Component<Props, State> {
+  componentDidMount() {
+    this.props.isLogged ? this.setState({logout: STRINGS.SIDE_MENU.LOGOUT}) : this.setState({logout: STRINGS.SIDE_MENU.LOGIN});
+  }
+  componentWillReceiveProps = (nextProps) => {
+    this.props.isLogged ? this.setState({logout: STRINGS.SIDE_MENU.LOGOUT}) : this.setState({logout: STRINGS.SIDE_MENU.LOGIN});
+  }
+
+  constructor(props: Props) {
```
Comment (falseinput) [inline src/javascripts/pages/side-menu/side-menu.js]: Constructor always go at the top of class

File: src/javascripts/pages/side-menu/side-menu.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+};
+
+type State = {
+  logout: String,
+};
+
+function mapDispatchToProps(dispatch) {
+  return {
+    toggleLoggedIn: function (data) {
+      return dispatch(userActions.actions.toggleLoggedIn(data));
```
Comment (falseinput) [inline src/javascripts/pages/side-menu/side-menu.js]: No neeed to return. You only need to call dispatch

File: src/javascripts/pages/side-menu/side-menu.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        console.log(err);
+      });
+  }
+
+  render() {
+    return (
+      <View style={styles.container}>
+        <ScrollView>
+          <View>
+            {/* Future */}
```
Comment (falseinput) [inline src/javascripts/pages/side-menu/side-menu.js]: Is there a future for us anyway...?

File: src/javascripts/pages/side-menu/side-menu.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+function mapDispatchToProps(dispatch) {
+  return {
+    toggleLoggedIn: function (data) {
+      return dispatch(userActions.actions.toggleLoggedIn(data));
+    }
+  };
+}
+
+class SideMenu extends React.Component<Props, State> {
+  componentDidMount() {
```
Comment (falseinput) [inline src/javascripts/pages/side-menu/side-menu.js]: I think that there is no need for local state in this component at all. Just check props in render method and return specific string.

File: src/javascripts/pages/side-menu/side-menu.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+      </View>
+    );
+  }
+}
+
+SideMenu.propTypes = {
+  navigation: PropTypes.object
+};
+
+export default SideMenu;
```
Comment (falseinput) [inline src/javascripts/pages/side-menu/side-menu.js]: It won't work this way. If you want to suscribe to store (via mapStateToProps) or change store data (via mapDispatchToProps) you need to:
a) `import { connect } from 'react-redux';`
b) `export default connect(mapStateToProps, mapDispatchToProps)(SideMenu) `

To make this work add also mapStateToProps function (see how it's done in diffrent coponents)

File: src/javascripts/pages/side-menu/side-menu.js (hunk 5 above)
Comment (Cezikos) [reply src/javascripts/pages/side-menu/side-menu.js]: Indeed

**Pull request merged**